engine/lib/phx/src/window/winit_windows.rs

```diff
@@ -0,0 +1,359 @@
+use std::sync::atomic::Ordering;
+
+use glutin::config::ConfigTemplateBuilder;
+use glutin::context::{ContextApi, ContextAttributesBuilder, GlProfile, Version};
+use glutin::display::GetGlDisplay;
+use glutin::prelude::{GlConfig, GlDisplay};
+use glutin_winit::DisplayBuilder;
+use hashbrown::HashMap;
+use raw_window_handle::HasRawWindowHandle;
+use tracing::{debug, error, info, warn};
+use winit::{
+    dpi::{LogicalSize, PhysicalPosition},
+    monitor::MonitorHandle,
+};
+
+use super::{CursorGrabMode, Window, WindowMode, WindowPosition, WindowResolution, WinitWindow};
+
+/// A resource which contains [`winit`] library windows.
+#[derive(Debug)]
+pub struct WinitWindows {
+    gl_version_major: u8,
+    gl_version_minor: u8,
+
+    /// Stores [`winit`] windows by window identifier.
+    pub windows: HashMap<winit::window::WindowId, WinitWindow>,
+
+    // Some winit functions, such as `set_window_icon` can only be used from the main thread. If
+    // they are used in another thread, the app will hang. This marker ensures `WinitWindows` is
+    // only ever accessed with LTR's non-send functions.
+    _not_send_sync: core::marker::PhantomData<*const ()>,
+}
+
+impl WinitWindows {
+    /// Create WinitWindows manager for specified GL version
+    pub fn new(gl_version_major: u8, gl_version_minor: u8) -> Self {
+        Self {
+            gl_version_major,
+            gl_version_minor,
+            windows: Default::default(),
+            _not_send_sync: Default::default(),
+        }
+    }
+
+    /// Creates a `winit` window and associates it with our entity.
+    pub fn create_window(
+        &mut self,
+        event_loop: &winit::event_loop::EventLoopWindowTarget<()>,
+        window: &Window,
+    ) -> winit::window::WindowId {
+        info!("Create new window: {}", window.title);
+
+        let mut winit_window_builder = winit::window::WindowBuilder::new();
+
+        // Due to a UIA limitation, winit windows need to be invisible for the
+        // AccessKit adapter is initialized.
```

Comment:
Sorry, not sure I understand this comment. What does UIA mean? What is AccessKit?

Reply:
Copy paste from bevy. Rephrased.